x/evm/keeper/grpc_query.go

```diff
@@ -351,3 +352,29 @@ func (k Keeper) StaticCall(c context.Context, req *types.QueryStaticCallRequest)
 
 	return nil, nil
 }
+
+// EthCall implements eth_call rpc api.
+func (k Keeper) EthCall(c context.Context, args *types.CallArgs) (*types.MsgEthereumTxResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	k.WithContext(ctx)
+
+	msg := args.ToMessage()
```

Comment:
validate msg fields using `ValidateBasic`

Reply:
Added address length validation in the `ToMessage`.